tests/foreman/cli/test_discoveredhost.py

```diff
@@ -14,50 +14,142 @@
+        {
+            'id': discovered_host.id,
+            'hostgroup-id': discovered_host.hostgroup.id,
+            'organization-id': discovered_host.organization.id,
+            'location-id': discovered_host.location.id,
+        }
+    )
+    assert 'Host created' in result[0]['message']
+    host = sat.api.Host().search(query={"search": f'id={discovered_host.id}'})[0]
+    assert host
+    wait_for(
+        lambda: host.read().build_status_label != 'Pending installation',
+        timeout=1500,
+        delay=10,
+    )
+    host = host.read()
+    assert host.build_status_label == 'Installed'
+
+    @request.addfinalizer
+    def _finalize():
+        host.delete()
+        assert not sat.api.Host().search(query={"search": f'id={discovered_host.id}'})
+
+
+@pytest.mark.e2e
+@pytest.mark.on_premises_provisioning
+@pytest.mark.parametrize('module_provisioning_sat', ['discovery'], indirect=True)
+@pytest.mark.parametrize('pxe_loader', ['bios', 'uefi'], indirect=True)
+@pytest.mark.rhel_ver_match('8')
+def test_positive_provision_pxe_less_host_cli(
+    module_discovery_sat,
+    pxeless_discovery_host,
+    module_provisioning_rhel_content,
+    provisioning_hostgroup,
+    request,
+):
+    """Provision a pxe-less discovered hosts with cli
+
+    :id: e75ee13a-9edc-4182-b02a-6b106a459751
+
+    :Setup: Provisioning should be configured and a host should be
+        discovered via cli
+
+    :expectedresults: Host should be provisioned successfully
+
+    :CaseImportance: Critical
+    """
+    sat = module_discovery_sat.sat
+    pxeless_discovery_host.power_control(ensure=False)
+    mac = pxeless_discovery_host._broker_args['provisioning_nic_mac_addr']
+
+    wait_for(
+        lambda: sat.api.DiscoveredHost().search(query={'mac': mac}) != [],
+        timeout=240,
+        delay=20,
+    )
+    discovered_host = sat.api.DiscoveredHost().search(query={'mac': mac})[0]
+    discovered_host.hostgroup = provisioning_hostgroup
+    discovered_host.location = provisioning_hostgroup.location[0]
+    discovered_host.organization = provisioning_hostgroup.organization[0]
+    discovered_host.build = True
+    result = sat.cli.DiscoveredHost.provision(
+        {
+            'id': discovered_host.id,
+            'hostgroup-id': discovered_host.hostgroup.id,
+            'organization-id': discovered_host.organization.id,
+            'location-id': discovered_host.location.id,
+        }
+    )
+    assert 'Host created' in result[0]['message']
+    host = sat.api.Host().search(query={"search": f'id={discovered_host.id}'})[0]
+    assert host
+    wait_for(
+        lambda: host.read().build_status_label != 'Pending installation',
+        timeout=1500,
+        delay=10,
+    )
+    host = host.read()
+    assert host.build_status_label == 'Installed'
+
+    @request.addfinalizer
+    def _finalize():
+        host.delete()
+        assert not sat.api.Host().search(query={"search": f'id={discovered_host.id}'})
```

Comment:
Can we move this finalizer in the middle of the test, instead of bottom?

Reply:
Why do you feel it is in the wrong place?